### PR TITLE
[inductor] [cpp] fix max-autotune for single-thread dynamic shapes

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #136419
* #136518
* __->__ #136418

Fixes the compilation error of max-autotune for `maml_omniglot` (AMP and FP32) and `soft_actor_critic` (AMP) in Torchbench for single-thread dynamic shapes case:

```
/tmp/torchinductor_user/uv/cuvq6wenwp7us423onuvntkfx4cspmagha5beiknob7tiebzhupa.cpp: In function ‘void kernel(const bfloat16*, const bfloat16*, const bfloat16*, bfloat16*, int64_t)’:
/tmp/torchinductor_user/uv/cuvq6wenwp7us423onuvntkfx4cspmagha5beiknob7tiebzhupa.cpp:279:41: error: the value of ‘Mr_blocks’ is not usable in a constant expression
  279 |         constexpr int64_t m_block_end = Mr_blocks;
      |                                         ^~~~~~~~~
/tmp/torchinductor_user/uv/cuvq6wenwp7us423onuvntkfx4cspmagha5beiknob7tiebzhupa.cpp:237:19: note: ‘Mr_blocks’ was not initialized with a constant expression
  237 |     const int64_t Mr_blocks = (M + Mr - 1) / Mr;
      |                   ^~~~~~~~~
```

The PR also updates the UT to add a test for `BS`=512 in single thread.
The previous case has `BS`=1024 equal to the `K` and `N` value. The generated code does not have symbolic shapes thus fails to capture the above issue.
By adding a case of `BS`=512, the generated code will have symbolic shape for the M dim and is able to reproduce the issue that this PR is addressing.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/inductor/test_cpu_select_algorithm.py [hunk 1]
```diff
@@ -1396,7 +1396,13 @@ def forward(self, x):
     @torch.no_grad
     @unittest.skipIf(not TEST_MKL, "Test requires MKL")
     @set_num_threads(1)
-    @parametrize("batch_size", (1024,))
+    @parametrize(
+        "batch_size",
+        (
+            512,
+            1024,
```
Comment (jgong5) [inline test/inductor/test_cpu_select_algorithm.py:1403]: Perhaps just using `512` would be good enough? I'm concerned about the added test time...

**Review by jgong5 — APPROVED** [review]

File: test/inductor/test_cpu_select_algorithm.py (hunk 1 above)
Comment (chunyuan-w) [reply test/inductor/test_cpu_select_algorithm.py]: Changed to just keep `512`.
